src/shell.rs

```diff
@@ -148,19 +154,41 @@ impl MessageInfo {
             verbosity,
             stdout_needs_erase: false,
             stderr_needs_erase: false,
+            cross_debug: None,
+        }
+    }
+
+    /// Check if env var `CROSS_DEBUG` is set
+    pub fn cross_debug(&mut self) -> bool {
+        if let Some(b) = self.cross_debug {
+            b
+        } else if let Ok(b) = std::env::var("CROSS_DEBUG")
+            .as_deref()
+            .map(bool_from_envvar)
+        {
+            self.cross_debug = Some(b);
+            b
+        } else {
+            self.cross_debug = Some(false);
+            false
         }
     }
```

Comment:
I think this is a better interface, no way to do it wrong accidentally. Would be nice to use `OnceCell` or `LazyLock` though

Reply:
If we go with a dedicated `cross_debug`, it should belong in the constructor, rather than being lazily evaluated. In that case, a static method which is then used by the constructor is a good idea, rather than have an option and doing the above every single time we print a message. I think this should be evaluated in `create`, and required as an argument in `new`. I do like decoupling the verbosity level from the debug info, since this was different behavior from `cargo`.

Reply:
the problem is that we call `MessageInfo::new` in a few places where it doesn't really make sense to check for the env var.

I can't think of a better way to solve this. I'll implement it in the constructor and we can see if it works out or if there is a better way

Reply:
Yeah no it doesn't really work out, with this I have to add the env getter to all invocations of `MessageInfo::new`